### PR TITLE
Remove lifetime restriction in file reader

### PR DESCRIPTION
**Pull request by sunchao**
Currently there's borrowing relation between FileReader and
RowGroupReader, due to the sharing of metadata. This restriction is not
necessary as row group reader should be allowed to out-live file reader.
This removes it by replacing the metadata with RC (reference-counting)
values.

### CODE REVIEW

File: src/file/reader.rs [hunk 1]
```diff
@@ -21,7 +21,11 @@ use std::rc::Rc;
 
 use basic::{Type, Compression, Encoding};
 use errors::{Result, ParquetError};
-use file::metadata::{RowGroupMetaData, FileMetaData, ParquetMetaData};
+use file::metadata::{
```
Comment (sadikovi) [inline src/file/reader.rs]: How do you feel about using `use file::metadata::*` here?

Comment (sunchao) [reply src/file/reader.rs]: Yes. Makes sense. I'll replace with star-import.

**Comment by sunchao** [comment]
I think this will solve the lifetime issue for the global iterator (since there will no longer be lifetime parameter for `RowGroupReader`), but haven't tried it yet...

**Comment by sunchao** [comment]
Merged. Thanks @sadikovi for the review!

**Comment by sadikovi** [comment]
All good! Great to see this update.

**Pull request merged**